text/0000-fabric-x-next-generation.md

```diff
@@ -0,0 +1,365 @@
+
+**BFT Properties of Arma:**
+Arma's BFT properties (safety and liveness) are derived from the underlying BFT consensus on headers and the mechanisms ensuring batch availability and primary rotation. $F+1$ BAFs ensure batch persistence before ordering, and the complaint mechanism handles faulty or censoring primaries.
+
+## 6. Transaction Structure
+The transaction structure is flattened to reduce serialization/deserialization overhead and complexity compared to Fabric's nested structures.
+
+```protobuf
+// Tx represents a transaction in the blockchain.
+message Tx {
+  string id = 1; // The transaction ID.
+  repeated TxNamespace namespaces = 2; // Namespaces associated with the transaction.
+  repeated bytes signatures = 3; // Signature per namespace.
+}
+
+// TxNamespace represents a namespace within a transaction.
+message TxNamespace {
+  string ns_id = 1; // The namespace ID.
+  bytes ns_version = 2; // The version of the namespace.
+  repeated Read reads_only = 3; // List of read-only operations within the namespace.
+  repeated ReadWrite read_writes = 4; // List of read-write operations within the namespace.
+  repeated Write blind_writes = 5; // List of blind write operations within the namespace.
+}
+
+// Read represents a read operation.
+message Read {
+  bytes key = 1; // The key being read.
+  bytes version = 2; // The version of the key being read.
+}
+
+// ReadWrite represents a read-write operation.
+message ReadWrite {
+  bytes key = 1; // The key involved in the read-write operation.
+                 // If 'key' is empty, it's constructed using the transaction ID
+                 // and the operation's index within the transaction.
+  bytes version = 2; // The version of the key being read and written.
+  bytes value = 3; // The value associated with the key being written.
+}
+
+// Write represents a write operation.
+message Write {
+  bytes key = 1; // The key being written.
+                 // If 'key' is empty, it's constructed using the transaction ID
+                 // and the operation's index within the transaction.
+  bytes value = 2; // The value associated with the key being written.
+}
+```
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+* **Increased Operational Complexity:** Managing a distributed system of microservices (Fabric-X peer and orderer components, distributed database per organization) is inherently more complex than managing monolithic Fabric nodes. This requires more sophisticated deployment (e.g., Kubernetes), monitoring, and orchestration tools, and potentially specialized operational skills.
+* **Backward Incompatibility:** As stated, Fabric-X is not backward compatible with existing Hyperledger Fabric versions. This means current Fabric users cannot simply upgrade; they would need to adopt a new implementation and potentially re-architect applications. This necessitates the separate repository approach to avoid disrupting the stable Fabric user base.
```

Comment:
> and potentially re-architect applications

Well, I think you underestimate the pain and onus that will be imposed on Fabric users, who will really think that Fabric-X is an evolution without properly understanding that this is just a spin-off for one particular use case. 

> This necessitates the separate repository approach to avoid disrupting the stable Fabric user base.

I re-read the updated RFC, and I am still very convinced that this mandates going through a proper project submission, for exactly this reason, of not disrupting the Fabric ecosystem. I bet people will get confused thinking this is somewhat compatible with whatever they already have.

Reply:
It is not hard to convey the differences between Fabric and Fabric-X through documentation, webinar, community calls, and technical workshop. If your major concern is the confusion, it can certainily be addressed.